### ISSUE TITLE
Elasticsearch 6.5 works with Java 11 as well

### ISSUE THREAD

**Issue by jiripospisil** [Homebrew/homebrew-core#34360]
- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?

Since Elasticsearch 6.5 it's possible to use Java 11 as well as Java 8. The [current formula](https://github.com/Homebrew/homebrew-core/blob/38ebd83aeeaa9b42033b1c104813ed1c4a4d6cd9/Formula/elasticsearch.rb#L14) however requires Java 8 and won't let me proceed with the installation. You can see the whole matrix [here](https://www.elastic.co/support/matrix#matrix_jvm).

```
$ java -version
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)

$ brew info elasticsearch
elasticsearch: stable 6.5.0, HEAD
(...)

$ brew install elasticsearch
elasticsearch: Java 1.8 is required to install this formula.
JavaRequirement unsatisfied!
You can install with Homebrew Cask:
 brew cask install homebrew/cask-versions/java8
You can download from:
 https://www.oracle.com/technetwork/java/javase/downloads/index.html
Error: An unsatisfied requirement failed this build.
```

**Comment by MikeMcQuaid** [comment]
Could you try and open a pull request to update this @jiripospisil? [This document](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request) should help and we're happy to walk you through anything else.

Thanks!

**Comment by pemontto** [comment]
Same with Logstash

```
$ brew install logstash
logstash: Java 1.8 is required to install this formula.
JavaRequirement unsatisfied!
You can install with Homebrew Cask:
 brew cask install homebrew/cask-versions/java8
You can download from:
 https://www.oracle.com/technetwork/java/javase/downloads/index.html
Error: An unsatisfied requirement failed this build.
```

**Comment by jiripospisil** [comment]
> Could you try and open a pull request to update this @jiripospisil? [This document](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request) should help and we're happy to walk you through anything else.

@MikeMcQuaid Absolutely, however, I'm not sure what's the syntax for `depends_on` to say "Require Java 1.8 or Java 11 but nothing in between or outside the range." As far as I know a semver expression such as `1.8 || 11` is not supported. Any hints? Is there maybe a formula with similar requirements (I haven't found any)?

**Comment by MikeMcQuaid** [comment]
@jiripospisil Correct, we don't support that syntax. My suggestion in that case would be to pick one.

**Comment by jiripospisil** [comment]
@MikeMcQuaid I think that would cause quite a lot of problems (and opened issues) down the road because both Java 8 and 11 are LTS releases and going to be supported for a number of years.

![image](https://user-images.githubusercontent.com/20820/49164835-3819e100-f330-11e8-9772-e2d58e9b2823.png)

Is support for more of semver planned? Given that you've gone so far without it maybe it wouldn't have to be full support but rather just the "or" for now.

**Comment by MikeMcQuaid** [comment]
@jiripospisil These version requirements are a fairly thin wrapper around `/usr/libexec/java_home`. It's never impossible to present a version requirement as either a specific version or greater than or equal to a specific version. There is no need for us to add the "or"; again I'd rather we picked one and declared that's the version we support. Until recently we've always relied on people having the newer version anyway (and require this with e.g. Xcode, XQuartz, all dependencies). In this case, then, this should be either `11` or `11+`.

**Comment by apjanke** [comment]
> For the moment to get it working I just changed my logstash.rb/elasticsearch.rb in /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula to
depends_on :java => "1.8+"

This isn't a solution to incorporate into the main `elasticsearch` formula, because on systems with just Java 9 and/or 10 installed, it'll pick an incompatible one.

I'd say leave it as is for now, since nothing is really broken, and that won't cause users with existing installs to have to install another JDK version to re-satisfy the dependency. And most of the formulae in core that have exact Java version requirements require `1.8`; I don't see any that require `11+` yet.

And for logstash specifically, we should stick with `1.8`, because that [compatibility matrix](https://www.elastic.co/support/matrix#matrix_jvm) says only Java 1.8 is supported, not Java 11.
